CompressImagesFunction/CompressImages.cs

```diff
@@ -134,7 +133,53 @@ public static bool Run(CompressimagesParameters parameters, ICollector<CompressI
             repo.Reset(ResetMode.Mixed, repo.Head.Tip);
 
             // optimize images
-            var imagePaths = ImageQuery.FindImages(parameters.LocalPath, repoConfiguration);
+            string[] imagePaths;
+            List<string> addedOrModifiedImagePaths = new List<string>();
+            List<string> deletedImagePaths = new List<string>();
+            if (parameters.IsRebase)
+            {
+                var refspec = string.Format("{0}:{0}", KnownGitHubs.BranchName);
+                Commands.Fetch(repo, "origin", new List<string> { refspec }, null, "fetch");
+
+                var diff = repo.Diff.Compare<TreeChanges>(repo.Branches[KnownGitHubs.BranchName].Commits.ElementAt(1).Tree, repo.Head.Tip.Tree);
+
+                if (diff == null)
```

Comment:
i think we should do something to validate that the head commit in this branch is actually made by imgbot.
I feel that it is important that we never modify anyone else's commits and we only amend commits made by the bot.

This is the scenario I am thinking about:
1. Imgbot opens a branch and adds a commit
2. User adds a new commit on top of imgbot's commit
3. This rebase runs and amends the user's commit